### PR TITLE
Tutorial for debugging a failing test using docker

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

Here is a new tutorial for debugging a failing test.  I wrote this because while this may seem trivial to some it seems useful to help anyone who is trying to debug a failing test with a tutorial of how to use docker in a similar way to our CI system.

### CODE REVIEW

File: doc/test_debugging/test_debugging_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
+Debugging Tests
+================
+
+How to debug when a test is failing.
+
+**Note:** *This is not meant as an exaustive tutorial on software testing, instead this focueses on methods that will help you debug tests in moveit or similar ros projects.*
```
Comment (rhaschke) [inline doc/test_debugging/test_debugging_tutorial.rst:6]:
```suggestion
**Note:** *This is not meant as an exhaustive tutorial on software testing, instead this focuses on methods that will help you debug tests in MoveIt or similar ROS projects.*
```

File: doc/test_debugging/test_debugging_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+Getting Started
+---------------
+If you haven't already done so, make sure you've completed the steps in `Getting Started <../getting_started/getting_started.html>`_.
+
+CI Failures
+-----------
+
+Our CI runs in travis and uses scripts found in the `moveit_ci repo <https://github.com/ros-planning/moveit_ci.git>`.  These test build nad run various tests in various enviroments.  Often something that works locally won't work in CI in a different enviroment.  To troubleshoot a failure from CI it is useful to use docker to run in the same enviroment.
```
Comment (rhaschke) [inline doc/test_debugging/test_debugging_tutorial.rst:15]:
```suggestion
Our CI runs in Travis and uses scripts found in the `moveit_ci repo <https://github.com/ros-planning/moveit_ci.git>`.  These tests build and run various tests in various environments.  Often something that works locally won't work in CI in a different environment.  To troubleshoot a failure from CI it is useful to use docker to run in the same environment.
```

File: doc/test_debugging/test_debugging_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  apt-get update
+  apt-get dist-upgrade
+  apt-get install -y python-catkin-tools ssh git gdb valgrind vim
+
+Then you'll want to set enviroment variables important for your test.  In my example I'm using a kinetic enviroment so I set that here.
+
+  ROS_DISTRO=kinetic
+
+Lastly, you'll want to create the folder structure for yoru ros enviroment:
```
Comment (rhaschke) [inline doc/test_debugging/test_debugging_tutorial.rst:45]:
```suggestion
Lastly, you'll want to create the folder structure for your ROS environment:
```

File: doc/test_debugging/test_debugging_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+Setup Enviroment
+----------------
+
+The first thing you should do is update debians and install tools you'll need for debugging.  The update is important because that is what we do in CI.
+
+  apt-get update
+  apt-get dist-upgrade
+  apt-get install -y python-catkin-tools ssh git gdb valgrind vim
+
+Then you'll want to set enviroment variables important for your test.  In my example I'm using a kinetic enviroment so I set that here.
```
Comment (rhaschke) [inline doc/test_debugging/test_debugging_tutorial.rst:41]:
```suggestion
Then you'll want to set environment variables important for your test.  In my example, I'm using a kinetic environment so I set that here.
```

File: doc/test_debugging/test_debugging_tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+  ROS_DISTRO=kinetic
```
Comment (rhaschke) [inline doc/test_debugging/test_debugging_tutorial.rst:43]: This should be defined within the ROS container by default.

**Comment by tylerjw** [comment]
@rhaschke thank you for the review.  I will refactor this pointing at the moveit_ci readme with some additional notes.

**Comment by felixvd** [comment]
> I'm not sure, why you essentially repeat all the instructions from travis.sh here.
> There is already a guide how to run the Travis script locally for testing [here](https://github.com/ros-planning/moveit_ci#running-locally-for-testing). 

Maybe he (like me) had no idea that it existed. It's not mentioned in the [Contributing](https://moveit.ros.org/documentation/contributing/#:~:text=You%20can%20contribute%20greatly%20to,community's%20official%20question%2Danswer%20forum.&text=In%20Interesting%20tags%20text%20field,like%20to%20get%20notifications%20about).) section on the website, the [Docker instructions](https://moveit.ros.org/install/docker/) or [our only test tutorial](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html). I've seen the package but didn't assume it had instructions for debugging your PRs. This PR also has helpful concrete steps and general debugging instructions, while the moveit_ci instructions require you to be familiar with more concepts (including what "Travis" and "CI" is to start with).

**Review by felixvd — COMMENTED** [review]
I think this is great and very valuable, but it should probably live in the [Integration/Unit Tests tutorial](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html), or can even complete/replace it.

**Comment by davetcoleman** [comment]
@felixvd I thought your feedback about merging with the existing test tutorial was smart, but after trying to merge the two, I think it makes for a very unweildy tutorial. I'd rather leave as-is

You also left a lot of "nice to have" feedback, but I think @tylerjw 's work here is good as-is, and follow up PRs can make it better. Better to have some documentation than non-existent documentation on this matter.

**Review by davetcoleman — APPROVED** [review]

**Comment by davetcoleman** [comment]
Retriggering the new Travis.com vs .org

File: doc/test_debugging/test_debugging_tutorial.rst [hunk 6]
```diff
@@ -0,0 +1,101 @@
+Debugging Tests
+================
```
Comment (davetcoleman) [inline doc/test_debugging/test_debugging_tutorial.rst:2]:
```suggestion
===============
```

File: doc/test_debugging/test_debugging_tutorial.rst [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+CI Failures
+-----------
+
+Our CI runs in Travis and uses scripts found in the `moveit_ci repo <https://github.com/ros-planning/moveit_ci.git>`.  These tests build and run various tests in various environments.  Often something that works locally won't work in CI in a different environment.  To troubleshoot a failure from CI it is useful to use docker to run in the same environment.
+
+For troubleshooting a specific travis test it is helpful to look at the .travis.yml config file and test output to understand what environment variables are being set in your test.
+
+Launch Docker Environment
+-----------------------
```
Comment (davetcoleman) [inline doc/test_debugging/test_debugging_tutorial.rst:20]:
```suggestion
-------------------------
```

File: doc/test_debugging/test_debugging_tutorial.rst [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  docker pull moveit/moveit:kinetic-ci
+  docker run -it moveit/moveit:kinetic-ci /bin/bash
+
+To start docker container for melodic:
+
+  docker pull moveit/moveit:melodic-ci
+  docker run -it moveit/moveit:melodic-ci /bin/bash
+
+Setup Environment
+----------------
```
Comment (davetcoleman) [inline doc/test_debugging/test_debugging_tutorial.rst:33]:
```suggestion
-----------------
```

**Review by davetcoleman — COMMENTED** [review]
Fixing travis issues.

**Pull request merged**